### ISSUE TITLE
concourse not compatible with ssh-keygen new ver.

### ISSUE THREAD

**Issue by kwladyka** [concourse/concourse#2949]
## Bug Report

It is already reported, but closed while it shouldn't be. I add here more details.
https://github.com/concourse/concourse/issues/2590

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x8ab5dd]

goroutine 1 [running]:
github.com/concourse/fly/commands.GetExpirationFromString(0xc00016ff40, 0xc0001b3410, 0xc00004bd60)
/tmp/build/9674af12/concourse/src/github.com/concourse/fly/commands/targets.go:59 +0x7d
github.com/concourse/fly/commands.(*TargetsCommand).Execute(0xe049b0, 0xc0001c4140, 0x0, 0x1, 0xe049b0, 0x1)
/tmp/build/9674af12/concourse/src/github.com/concourse/fly/commands/targets.go:33 +0x57f
github.com/jessevdk/go-flags.(*Parser).ParseArgs(0xc00005ef00, 0xc00000a090, 0x1, 0x1, 0x90c580, 0xc0001a95b0, 0xc00004bf38, 0x8a9237, 0xc0001026c0)
/tmp/build/9674af12/concourse/src/github.com/jessevdk/go-flags/parser.go:316 +0x800
github.com/jessevdk/go-flags.(*Parser).Parse(0xc00005ef00, 0x9d67d6, 0x8, 0xc0000fe780, 0x0, 0x0)
/tmp/build/9674af12/concourse/src/github.com/jessevdk/go-flags/parser.go:186 +0x71
main.main()
/tmp/build/9674af12/concourse/src/github.com/concourse/fly/main.go:24 +0x101
```

Bug still exist with 4.2.2.

## Steps to Reproduce

Run ssh-keygen on updated system like OS X, which use the newest versions. I can send files from my computer if you need to.
```
ssh-keygen -t rsa -f host-key  -N ''
ssh-keygen -t rsa -f worker-key  -N ''
ssh-keygen -t rsa -f session-signing-key  -N ''
```

The reason of the issue:
https://www.openssh.com/txt/release-7.8

## Expected Results

Concourse work with new certificates files.

## Actual Results

It fail to start workers.

## Additional Context

Here is workaround for this bug:
Add `-e PEM` to `ssh-keygen`, so commands should look like that:
`ssh-keygen -e PEM -t rsa -f host-key  -N ''`

This will generate keys in old style.

I found it here:
https://github.com/concourse/concourse-docker/blob/master/generate-keys.sh

## Version Info

* Concourse version: 4.2.2

**Comment by xtremerui** [comment]
Quote from the openssh release note

> ssh-keygen(1): write OpenSSH format private keys by default
   instead of using OpenSSL's PEM format. The OpenSSH format,
   supported in OpenSSH releases since 2014 and described in the
   PROTOCOL.key file in the source distribution, offers substantially
   better protection against offline password guessing and supports
   key comments in private keys. If necessary, it is possible to write
   old PEM-style keys by adding "-m PEM" to ssh-keygen's arguments
   when generating or updating a key. 

Should we keep using PEM or switch to OpenSSH format?

**Comment by kwladyka** [comment]
1) Both if we can do it simple
2) New if we can (OpenSSH)

**Comment by marco-m** [comment]
I think this can be closed due to #3330

**Comment by vito** [comment]
@marco-m yep, thanks for the assist! there were quite a few issues in different places for this one, heh
